baybe/surrogates/gaussian_process/presets/default.py

```diff
@@ -22,31 +23,96 @@
 class DefaultKernelFactory(KernelFactory):
     """A factory providing the default kernel for Gaussian process surrogates.
 
+    This is taking the low and high dimensional limits of
+    :class:`baybe.surrogates.gaussian_process.presets.default.EDBOKernelFactory`
+    and interpolates the prior moments linearly between them.
+    """
+
+    def __call__(  # noqa: D102
+        self, searchspace: SearchSpace, train_x: Tensor, train_y: Tensor
+    ) -> Kernel:
+        # See base class.
+
+        # Calculate effective dimensions, ignoring task parameters
+        n_task_parameters = len(
+            [p for p in searchspace.parameters if isinstance(p, TaskParameter)]
+        )
+        effective_dims = train_x.shape[-1] - n_task_parameters
+
+        # Fix dimensionality limits
+        LOW_D_LIM, HIGH_D_LIM = 8, 100
+
+        # Apply prior logic
+        if effective_dims < LOW_D_LIM:
+            # Low D priors
+            lengthscale_prior = GammaPrior(1.2, 1.1)
+            lengthscale_initial_value = 0.2
+            outputscale_prior = GammaPrior(5.0, 0.5)
+            outputscale_initial_value = 8.0
+        elif effective_dims > HIGH_D_LIM:
+            # High D priors
+            lengthscale_prior = GammaPrior(3.0, 1.0)
+            lengthscale_initial_value = 2.0
+            outputscale_prior = GammaPrior(5.0, 0.2)
+            outputscale_initial_value = 20.0
+        else:
+            # Interpolate prior moments linearly between low D and high D regime
+            interp_factor = (effective_dims - LOW_D_LIM) / (HIGH_D_LIM - LOW_D_LIM)
+
+            def _interp(a, b):
+                return a + (b - a) * interp_factor
+
+            lengthscale_prior = GammaPrior(_interp(1.2, 3.0), _interp(1.1, 1.0))
+            lengthscale_initial_value = _interp(0.2, 2.0)
+            outputscale_prior = GammaPrior(5.0, _interp(0.5, 0.2))
+            outputscale_initial_value = _interp(8.0, 20)
+
+        return ScaleKernel(
+            MaternKernel(
+                nu=2.5,
+                lengthscale_prior=lengthscale_prior,
+                lengthscale_initial_value=lengthscale_initial_value,
+            ),
+            outputscale_prior=outputscale_prior,
+            outputscale_initial_value=outputscale_initial_value,
+        )
+
+
+@define
+class EDBOKernelFactory(KernelFactory):
```

Comment:
not tested

Reply:
added now, please check
kernel factory was not tested preivously afaik because the default kernel fixture is set to a plain matern kernel not factory